packages/test/test-version-utils/src/versionUtils.ts

```diff
@@ -285,10 +285,41 @@ export function checkInstalled(requested: string) {
  */
 export const loadPackage = async (modulePath: string, pkg: string): Promise<any> => {
 	const pkgPath = path.join(modulePath, "node_modules", pkg);
-	const pkgJson: { main: string } = JSON.parse(
+	// Because we put legacy versions in a specific subfolder of node_modules (.legacy/<version>), we need to reimplement
+	// some of Node's module loading logic here.
+	// It would be ideal to remove the need for this duplication (e.g. by using node:module APIs instead) if possible.
+	const pkgJson: { main?: string; exports?: string | Record<string, string> } = JSON.parse(
 		readFileSync(path.join(pkgPath, "package.json"), { encoding: "utf8" }),
 	);
-	return import(pathToFileURL(path.join(pkgPath, pkgJson.main)).href);
+	// See: https://nodejs.org/docs/latest-v18.x/api/packages.html#package-entry-points
+	let primaryExport: string;
+	if (pkgJson.exports !== undefined) {
+		// See https://nodejs.org/docs/latest-v18.x/api/packages.html#conditional-exports for information on the spec
+		// if this assert fails.
+		// The v18 doc doesn't mention that export paths must start with ".", but the modern docs do:
+		// https://nodejs.org/api/packages.html#exports
+		for (const key of Object.keys(pkgJson.exports)) {
```

Comment:
we could implement this, but I kind of hate how complex this code is anyway (and that it's duplicating the job of Node's module loader).

Though I could see us using conditional exports in some of our packages, I don't think we'll need to do so at least for the packages we directly import via this function.